File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.js
class Solution {
    /**
     * @param {number[]} prices
     * @return {number}
     */
    maxProfit(prices) {
        let maxProfit = 0
        let minPrice = Infinity

        for(let sell of prices) {
            minPrice = Math.min(minPrice, sell)
            let profit = sell - minPrice
            maxProfit = Math.max(maxProfit, profit)
        }
        return maxProfit
    }
}

File: Data Structures & Algorithms/best-time-to-buy-and-sell-stock-ii/submission-0.go
func maxProfit(prices []int) int {
	n:= len(prices)
	if n<2 {
		return 0
	}

	leftBuy, rightSell := 0, 1
	profit := 0

	for rightSell < n {
		currPrice := prices[leftBuy]
		futurePrice := prices[rightSell]

		if currPrice < futurePrice {
			profit += futurePrice - currPrice
		}

		leftBuy = rightSell
		rightSell++
	}
	return profit
}
